### ISSUE TITLE
[BUG] The distributed bun packaged' code causes an error

### ISSUE THREAD

**Issue by bogeychan** [elysiajs/elysia#348]
Executing the following code (`bun run index.ts`) will result in an error caused by the distributed bun packaged' code:

### Code

```ts
// index.ts

import { TypeSystem } from '@sinclair/typebox/system';
import { Elysia } from 'elysia';

const BufferType = TypeSystem.Type<Buffer>(
  'BufferType',
  (_, value) => value instanceof Buffer
);

new Elysia()
  .post(
    '/',
    (ctx) => {
      const buffer = ctx.body;
      return { yay: 'yay' };
    },
    {
      async parse(ctx) {
        if (ctx.headers['content-type'] === 'application/json') {
          return Buffer.from(await ctx.request.arrayBuffer());
        }
      },
      body: BufferType()
    }
  )
  .listen(8080);
```

### Error

```bash
error: Preflight validation check failed to guard for the given schema
      at new W1 (/home/bogeychan/test/node_modules/elysia/dist/bun/index.js:2:14163)
      at new l$ (/home/bogeychan/test/node_modules/elysia/dist/bun/index.js:2:218979)
      at M0 (/home/bogeychan/test/node_modules/elysia/dist/bun/index.js:2:240103)
      at w1 (/home/bogeychan/test/node_modules/elysia/dist/bun/index.js:2:240654)
      at add (/home/bogeychan/test/node_modules/elysia/dist/bun/index.js:316:16246)
      at post (/home/bogeychan/test/node_modules/elysia/dist/bun/index.js:316:32322)
      at /home/bogeychan/test/index.ts:9:1
```

### Environment

- `elysia@0.7.30`
- `@sinclair/typebox@0.31.28`
- `bun@1.0.15`

### Cause

works as expected If I remove this line from `/home/bogeychan/test/node_modules/elysia/package.json`:

```diff
"exports": {
    ".": {
-      "bun": "./dist/bun/index.js",
      "node": "./dist/cjs/index.js",
      "require": "./dist/cjs/index.js",
      "import": "./dist/index.js",
      "default": "./dist/index.js"
    },
```

**Comment by sistematico** [comment]
Same issue here for this code:

```
import { Elysia, t } from "elysia";
import { mkdir } from "node:fs/promises";
import { db } from "@/model/database";
import WebSocket from 'ws';


const app = new Elysia({ prefix: "/upload" });
const ws = new WebSocket(`${Bun.env.WSS_URL}`);
const postsPath = Bun.env.POSTS_PATH 

app.post("*", async ({ headers: { content, uploader }, body: { files, fileNames, mimeTypes } }) => {
  const names = JSON.parse(String(fileNames));

  const post = await db.post.create({
    data: {
      content: content,
      authorId: uploader,
    }
  });

  if (!post) return "Error";

  const filePath = `${postsPath}/${uploader}/${post.id}`;
  const fpath = Bun.file(filePath);
  const webPath = `/posts/${uploader}/${post.id}`;
  
  if (!await fpath.exists()) await mkdir(filePath, { recursive: true });

  if (Array.isArray(files) && Array.isArray(names)) {
    for (let index = 0; index < files.length; index++) {
      const mediaFileName = String(names[index]);
      const mediaFile = files[index];

      const path = `${filePath}/${mediaFileName}`;
      const file = Bun.file(path);
      if (!await file.exists()) Bun.write(file, mediaFile);
      await db.mediaItem.create({ data: { uploaderId: Number(uploader), filePath: `${webPath}/${mediaFileName}`, postId: post.id } });
    }

    ws.send(JSON.stringify({ code: 1, message: "Novo post" }));

  } else if (files instanceof File && typeof fileNames === 'string') {
    const mediaFileName = String(names[0]);

    const path = `${filePath}/${mediaFileName}`;
    const file = Bun.file(path);
    if (!await file.exists()) Bun.write(file, files);

    await db.mediaItem.create({ data: { uploaderId: Number(uploader), filePath: `${webPath}/${mediaFileName}`, postId: post.id } });

    return "OK" + new Date();
  } else {
    return "ERROR";
  }

  ws.send(JSON.stringify({ code: 1, message: "Novo post" }));
  
  return "OK";
},
{
  body: t.Object({
    files: t.Array(t.Files()),
    fileNames: t.Object(t.String()),
    mimeTypes: t.Optional(t.Array(t.String()))
  }),
  headers: t.Object({
    content: t.Optional(t.String()),
    uploader: t.Numeric()
  })
});

export default app;
```

**Comment by SaltyAom** [comment]
If I remember correctly, I think this problem was fixed some months ago? @bogeychan

**Comment by bogeychan** [comment]
Yes, i can't reproduce it too. Looks like it has been fixed since ~1.0.0 release
